### PR TITLE
[typescript-migration] popper_component.jsx

### PR DESCRIPTION
**Pull request by yuki0410-dev**
## Description
**Linked issue**: #4700

**Changes** `popper_component.jsx` was migrated to ts

## Contribution checklist
* [x]  I have followed the [contributing guidelines](https://github.com/Hacker0x01/react-datepicker/blob/main/CONTRIBUTING.md).
* [x]  I have added sufficient test coverage for my changes.
* [x]  I have formatted my code with Prettier and checked for linting issues with ESLint for code readability.

### CODE REVIEW

File: src/popper_component.tsx [hunk 1]
```diff
@@ -1,35 +1,43 @@
...
-import React from "react";
-import PropTypes from "prop-types";
+import React, { createElement } from "react";
 import { FloatingArrow } from "@floating-ui/react";
+import type { ReferenceType, UseFloatingReturn } from "@floating-ui/react";
 import TabLoop from "./tab_loop";
 import Portal from "./portal";
 import withFloating from "./with_floating";
 
+interface PortalPorps
```
[the remark this answers is not in the document]
Comment (yuki0410-dev) [reply src/popper_component.tsx]: fixed :+1:

File: src/popper_component.tsx [hunk 2]
```diff
@@ -1,35 +1,43 @@
...
+  extends Omit<React.ComponentPropsWithoutRef<typeof Portal>, "children"> {}
+interface TabLoopPorps
+  extends Omit<React.ComponentPropsWithoutRef<typeof TabLoop>, "children"> {}
+
+interface PopperComponentProps<RT extends ReferenceType = ReferenceType>
+  extends PortalPorps,
+    TabLoopPorps {
```
[the remark this answers is not in the document]
Comment (yuki0410-dev) [reply src/popper_component.tsx]: fixed :+1:

**Comment by yuki0410-dev** [comment]
@martijnrusschen
If there are no other problems, I would be happy to merge them.

**Comment by martijnrusschen** [comment]
apologies, I'm on vacation this week and don't have the ability to review
or merge anything until Friday.

**Comment by yuki0410-dev** [comment]
Understood. Have a nice vacation.

**Comment by martijnrusschen** [comment]
Looks like there's a failing test, can you check?

**Comment by yuki0410-dev** [comment]
@martijnrusschen 
Rebased.

**Pull request merged**